genestack_client/genestack_connection.py

```diff
@@ -221,7 +220,10 @@ def __init__(self, connection, application_id=None):
         if len(self.application_id.split('/')) != 2:
             raise GenestackException('Invalid application ID, expect "{vendor}/{application}" got: %s' % self.application_id)
 
-    def __invoke(self, path, post_data):
+    def __invoke(self, path, post_data, trace=None):
+        self.connection.opener.addheaders = [('gs-extendSession', 'true')]
+        if trace:
+            self.connection.opener.addheaders.append(('gs-trace', 'true'))
         f = self.connection.open(path, post_data)
```

Comment:
It is not an error, this class is returned. It is file like object with info(headers) and  url.